### PR TITLE
[fx/profiler] provide a table of summary.

### PR DESCRIPTION
**Pull request by super-dainiu**
### What's new?
I create a `summary` method for `MetaInfoProp`. Hopefully, this will help @Cypher30 debug backward memory estimations.
```python
BATCH_SIZE = 2
DIM_IN = 4
DIM_HIDDEN = 16
DIM_OUT = 16
model = torch.nn.Sequential(
          torch.nn.Linear(DIM_IN, DIM_HIDDEN), 
          torch.nn.Linear(DIM_HIDDEN, DIM_OUT),
          )
input_sample = torch.rand(BATCH_SIZE, DIM_IN)
gm = symbolic_trace(model)
interp = MetaInfoProp(gm)
interp.run(input_sample)
print(interp.summary(format='kb'))    # don't panic if some statistics are 0.00 MB

======================================================
# output of above code is 
    Op type       Op    Forward FLOPs    Backward FLOPs    SAVE_FWD_IN    FWD_OUT    FWD_TMP    BWD_OUT    BWD_TMP
-----------  -------  ---------------  ----------------  -------------  ---------  ---------  ---------  ---------
placeholder  input_1          0 FLOPs           0 FLOPs          False    0.00 KB    0.00 KB    0.00 KB    0.00 KB
call_module       _0        128 FLOPs         288 FLOPs           True    0.12 KB    0.00 KB    0.34 KB    0.00 KB
call_module       _1        512 FLOPs       1,056 FLOPs           True    0.12 KB    0.00 KB    1.19 KB    0.00 KB
     output   output          0 FLOPs           0 FLOPs           True    0.00 KB    0.00 KB    0.00 KB    0.00 KB
```

### Warnings 
@Cypher30 
Obviously, `nn.Linear`'s `BWD_OUT` should be the same as the previous node's `FWD_OUT`.
image
It is fairly likely that `BWD_OUT` also includes a gradient for weight and bias.

### CODE REVIEW

File: colossalai/fx/passes/meta_info_prop.py [hunk 1]
```diff
@@ -236,3 +242,64 @@ def propagate(self, *args):
...
+        """
+        # https://github.com/pytorch/pytorch/blob/master/torch/fx/graph.py
+        try:
+            from tabulate import tabulate
+        except ImportError:
+            print("`print_tabular` relies on the library `tabulate`, "
+                  "which could not be found on this machine. Run `pip "
+                  "install tabulate` to install the library.")
+
+        assert self._is_proped, "Please call `interp.run(input)` before calling `interp.summary()`."
```
Comment (super-dainiu) [inline colossalai/fx/passes/meta_info_prop.py:260]: we need at least run meta once before getting a summary

File: colossalai/fx/passes/meta_info_prop.py [hunk 2]
```diff
@@ -236,3 +242,64 @@ def propagate(self, *args):
             Any: The value returned from executing the Module
         """
         return super().run(*args)
+
+    def summary(self, format: str = 'MB') -> str:
```
Comment (FrankLeeeee) [inline colossalai/fx/passes/meta_info_prop.py:246]: should it be `unit` instead of `format`?

Comment (super-dainiu) [reply colossalai/fx/passes/meta_info_prop.py:246]: yes

**Review by FrankLeeeee — APPROVED** [review]
